### ISSUE TITLE
Distinguish not required and null

### ISSUE THREAD

**Issue by steinymity** [cwacek/python-jsonschema-objects#180]
Consider a schema like this:

```json
{
    "type": "object",
    "required": [
        "result"
    ],
    "additionalProperties": false,
    "properties": {
        "result": {
            "title": "Response Data",
            "description": "Only present on success.",
            "type": [
                "string", "number", "object", "array", "boolean", "null"
            ]
        }
    }
}
```

When I instantiate such an object, the presence of `result` is not enforced since `null` is an accepted value. I.e. the empty object is accepted by pjs (but not by the `jsonschema` package).

Moreover, in above example, I have to list all possible types since pjs does not accept omitting `type`.

**Comment by cwacek** [comment]
Unfortunately, this schema is particularly problematic for something that's trying to generate class bindings, because you're effectively saying "this is required but can be anything, including nothing". Given that's the case, it's not actually possible to enforce requiredness.

**Comment by cwacek** [comment]
This also doesn't seem very practical. Can you outline how you'd expect this schema to be used?

**Comment by steinymity** [comment]
This is actually a snippet where I tried to implement Json-RPC. So, my intention was to enforce the presence of the `result` attribute without restricting it's type in any way since the [Json-RPC standard mandates that `result` must be present](https://www.jsonrpc.org/specification#response_object) in case of successful method execution.

As I indicated, I initially wanted to omit the `type` setting entirely, which should be fine, according to the standard.
